### PR TITLE
Add follow requests

### PR DESCRIPTION
**Pull request by michaviehauser**
Adds follow requests that have been requested by the community:

See #101 

Open for feedback!

Cheers,
Micha

### CODE REVIEW

**Comment by michaviehauser** [comment]
@overtrue could you please check PHPUnit needing PHP7.2 in the Scrutinizer build

Thanks!

File: src/Followable.php [hunk 1]
```diff
@@ -21,32 +21,93 @@
  */
 trait Followable
 {
+    /**
+     * @return bool
+     */
+    public function isPrivateUser()
```
Comment (overtrue) [inline src/Followable.php:27]: Thanks to the great PR, But I think this method may require a more appropriate name,  for example, `isFollowerNeedToBeReviewed` or `isFollowerNeedsReview`, `isFollowerNeedsApproval`? Sorry, I'm not good at English.

File: src/Followable.php [hunk 2]
```diff
@@ -21,32 +21,93 @@
...
+    }
+
+    /**
+     * @param \Illuminate\Database\Eloquent\Model|int $user
+     */
+    public function acceptFollowRequestFrom($user)
+    {
+        if ($user instanceof Model) {
+            $user = $user->getKey();
+        }
```
Comment (overtrue) [inline src/Followable.php:84]: `updateExistingPivot` has contains this logic.

File: src/Followable.php [hunk 3]
```diff
@@ -21,32 +21,93 @@
...
+    }
+
+    /**
+     * @param \Illuminate\Database\Eloquent\Model|int $user
+     */
+    public function rejectFollowRequestFrom($user)
+    {
+        if ($user instanceof Model) {
+            $user = $user->getKey();
+        }
```
Comment (overtrue) [inline src/Followable.php:72]: `detach` has contains this logic.

File: src/Followable.php [hunk 4]
```diff
@@ -21,32 +21,93 @@
...
+        if ($this->relationLoaded('followings')) {
+            return $this->followings
+                ->filter(function ($following) {
+                    return $following->pivot->accepted_at === null;
+                })
+                ->contains($user);
+        }
+
+        return $this->followings()
+            ->whereNull('accepted_at')
```
Comment (overtrue) [inline src/Followable.php:108]: maybe `->wherePivot('accepted_at', null)` better.

File: src/Followable.php [hunk 5]
```diff
@@ -62,10 +123,17 @@ public function isFollowing($user)
...
+            return $this->followings
+                ->filter(function ($following) {
+                    return $following->pivot->accepted_at !== null;
+                })
+                ->contains($user);
         }
 
-        return $this->followings()->where($this->getQualifiedKeyName(), $user)->exists();
+        return $this->followings()
+            ->whereNotNull('accepted_at')
```
Comment (overtrue) [inline src/Followable.php:134]: maybe `->wherePivot('accepted_at', '!=', null)` better.

File: src/Followable.php [hunk 6]
```diff
@@ -21,32 +21,93 @@
...
+        if ($user instanceof Model) {
+            $user = $user->getKey();
+        }
+
+        /* @var \Illuminate\Database\Eloquent\Model $this */
+        if ($this->relationLoaded('followings')) {
+            return $this->followings
+                ->filter(function ($following) {
+                    return $following->pivot->accepted_at === null;
+                })
```
Comment (overtrue) [inline src/Followable.php:103]:
```diff
- ->filter(function ($following) {
-            return $following->pivot->accepted_at === null;
- })
+ ->whereNull('pivot.accepted_at')
```

File: src/Followable.php [hunk 7]
```diff
@@ -62,10 +123,17 @@ public function isFollowing($user)
 
         /* @var \Illuminate\Database\Eloquent\Model $this */
         if ($this->relationLoaded('followings')) {
-            return $this->followings->contains($user);
+            return $this->followings
+                ->filter(function ($following) {
+                    return $following->pivot->accepted_at !== null;
+                })
```
Comment (overtrue) [inline src/Followable.php:129]:
```diff
- ->filter(function ($following) {
-            return $following->pivot->accepted_at !== null;
- })
+ ->whereNull('pivot.accepted_at')
```

File: src/Followable.php (hunk 1 above)
Comment (michaviehauser) [reply src/Followable.php:27]: Good input! Will update this

File: src/Followable.php (hunk 4 above)
Comment (michaviehauser) [reply src/Followable.php:108]: True, is more explicit

File: src/Followable.php (hunk 7 above)
Comment (michaviehauser) [reply src/Followable.php:129]: here it would need to be ->whereNotNull('pivot.accepted_at')

Comment (overtrue) [reply src/Followable.php]: Oh, I'm sorry! it is a typo.

**Comment by michaviehauser** [comment]
Thanks for the feedback!

**Comment by overtrue** [comment]
Thanks!

**Pull request merged**